frontend/src/components/Modal.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useRef } from "react";
+import { useNavigate } from "react-router-dom";
+
+import checkMark from "../assets/checkMarkIcon.svg";
+import closeIcon from "../assets/closeIcon.svg";
+import errorAlert from "../assets/errorAlert.svg";
+import questionMark from "../assets/questionMark.svg";
+import warningHand from "../assets/warningHand.svg";
+
+import { Button } from "./Button.tsx";
+import styles from "./Modal.module.css";
+
+type ModalProps = { isOpen: boolean; onClose: () => void; children?: React.ReactNode };
+
+export function Modal(props: ModalProps) {
+  const { isOpen, onClose, children } = props;
+
+  const drawerRef = useRef<HTMLDivElement>(null);
+  const closeButtonRef = useRef<HTMLImageElement>(null);
+  const modalWindowRef = useRef<HTMLDivElement>(null);
+  const canvasRef = useRef<HTMLDivElement>(null);
+
+  function handleMouseClick(e: React.MouseEvent<HTMLElement>) {
+    if (e.target === canvasRef.current || e.target === closeButtonRef.current) {
+      onClose();
+    }
+  }
+
+  useEffect(() => {
+    // Focus the container div automatically to listen for "Escape" key press
+    if (drawerRef && isOpen) {
+      if (drawerRef?.current) drawerRef.current.focus();
+    }
+  }, [drawerRef, isOpen]);
+
+  function handleKeyPress(event: React.KeyboardEvent<HTMLDivElement>) {
+    // Close modal when user clicks "Escape" key
+    if (event.key === "Escape") {
+      onClose();
+    }
+  }
+
+  return (
+    <div
+      className={styles.modalDrawer}
+      ref={drawerRef}
+      onKeyDown={handleKeyPress}
+      onClick={handleMouseClick}
+      tabIndex={-1}
+      role="presentation"
+    >
+      {isOpen && (
+        <div className={styles.popupCanvas} ref={canvasRef}>
+          <div className={styles.popupWindow} ref={modalWindowRef}>
+            <button
```

Comment:
The close icon is tab indexable but not keyboard interactable. Hitting enter or space doesn't trigger the closing. We likely need to extend the handleKey Press function. Escape  works